### PR TITLE
[react-native-cli] Update docs

### PR DESCRIPTION
**Pull request by martinbigio**
cc @mkonicek, @ide, @foghina

### CODE REVIEW

**Comment by martinbigio** [comment]
image

**Comment by martinbigio** [comment]
image

**Comment by martinbigio** [comment]
Should we keep documented how the old cli worked?

File: docs/RunningOnDeviceIOS.md [hunk 1]
```diff
@@ -28,19 +28,24 @@ You can also pack all the JavaScript code within the app itself. This way you ca
 1. Open `AwesomeApp/ios/AwesomeApp/AppDelegate.m`
 2. Follow the instructions for "OPTION 2":
   * Uncomment `jsCodeLocation = [[NSBundle mainBundle] ...`
-  * Run the `react-native bundle` command in terminal from the root directory of your app
+  * Run the `react-native bundle --platform <ios|android> --entry-file <path> --bundle-output <path> [--sourcemap-output path> --dev --minify]` command in terminal from the root directory of your app
```
Comment (frantic) [inline docs/RunningOnDeviceIOS.md]: After https://github.com/facebook/react-native/commit/be70e32de2c417a467af902e7f4e2a0058c8e171 there is no need to do that on iOS, the bundle is generated as part of Xcode build.

File: docs/RunningOnDeviceIOS.md [hunk 2]
```diff
@@ -28,19 +28,24 @@ You can also pack all the JavaScript code within the app itself. This way you ca
...
 
-* API is now `entry-file <path>` based instead of url based.
-* Need to specify which platform you're bundling for `--platform <ios|android>`.
-* Option `--out` has been renamed for `--bundle-output`.
-* Source maps are no longer automatically generated. Need to specify `--sourcemap-output <path>` 
+      $ react-native bundle --platform ios --entry-file index.ios.bundle --bundle-output ios/main.jsbundle --sourcemap-output ios/index.ios.map --dev false --minify true
+
+whereas the typical for android looks like:
+
+      $ react-native bundle --platform android --entry-file index.android.bundle --bundle-output android/app/src/main/assets/index.android.bundle --sourcemap-output android/index.android.map --dev false --minify true
```
Comment (frantic) [inline docs/RunningOnDeviceIOS.md]: Don't think Android needs this either. @foghina's new gradle plugin runs bundle command automatically.

**Comment by frantic** [comment]
I recommend you adding a different guide that explains CLI commands if you want to document them. Felix and me tried to make sure end used doesn't have to run anything, just normal Xcode/Gradle build.

However, this won't work for projects generated before v0.14. I should document this here.

**Comment by martinbigio** [comment]
Sounds good. I'l do that instead :)

**Pull request closed without merging**